akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -206,32 +210,38 @@ import akka.util.OptionVal
   }
 
   override val supervisorStrategy = classic.OneForOneStrategy(loggingEnabled = false) {
-    case TypedActorFailedException(cause) =>
-      // These have already been optionally logged by typed supervision
-      recordChildFailure(cause)
-      classic.SupervisorStrategy.Stop
     case ex =>
-      val isTypedActor = sender() match {
-        case afwc: ActorRefWithCell =>
-          afwc.underlying.props.producer.actorClass == classOf[ActorAdapter[_]]
+      ctx.setCurrentActorThread()
```

Comment:
Here it is needed because we use the logger?

Reply:
yes, the principle is that all entry points to message processing should set it. That's why it's in the around methods and supervision.